### ISSUE TITLE
versions:update-properties does not resolve X.Y-something-SNAPSHOT properly

### ISSUE THREAD

**Issue by joel-schaal** [mojohaus/versions-maven-plugin#412]
### Context
We have a library (*libfoo*) that has (at least) one feature branch and a master branch. The version in the pom is specified as follows:
* branch **master**: `1.0-SNAPSHOT`
* branch **featureA**: `1.0-featureA-SNAPSHOT`

Then we have an application (*appbar*) which also has feature branches:
* branch **master**: `2.0-SNAPSHOT`
* branch **featureA**: `2.0-featureA-SNAPSHOT`

The dependency between *appbar* and *libfoo* is defined through a property, as follow:
```
<project xmlns="http://maven.apache.org/POM/4.0.0"
 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.example</groupId>
  <artifactId>appbar</artifactId>
  <version>2.0-featureA-SNAPSHOT</version>
  
  <properties>
    <libfoo.version>1.0-featureA-SNAPSHOT</libfoo.version>
  </properties>
  
  <dependencies>
    <dependency>
      <groupId>com.example</groupId>
      <artifactId>libfoo</artifactId>
      <version>${libfoo.version}</version>
    </dependency>
  </dependencies>
</project>
```

### Issue
When using `mvn -U versions:update-properties` to replace the -SNAPSHOT with the actual latest corresponding version, we end up with a version from the master branch (ex:`1.0-1234`) instead of one from the feature branch (ex:`1.0-featureA-1235`)

### Workaround
If we define the version directly in the dependency:
```
<project xmlns="http://maven.apache.org/POM/4.0.0"
 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.example</groupId>
  <artifactId>appbar</artifactId>
  <version>2.0-SNAPSHOT</version>
  
  <dependencies>
    <dependency>
      <groupId>com.example</groupId>
      <artifactId>libfoo</artifactId>
      <version>1.0-SNAPSHOT</version>
    </dependency>
  </dependencies>
</project>
```
And if we use `mvn -U versions:use-releases` instead, then we end up with the appropriate version resolved. But this is a solution that does not scale well: if we have 10+ dependencies with the same version (ex: multi-module repository where each lib is a module), then at each version upgrade, the version should be adapted for each dependency. 

### TL;DR
- `versions:use-releases` resolves `1.0-something-SNAPSHOT` into `1.0-something-XXXXX`.
- `versions:update-properties` resolves `1.0-something-SNAPSHOT` into `1.0-XXXXX`.
- `versions:update-properties`  should behave like `versions:use-releases`.

**Comment by joel-schaal** [comment]
Up ?

**Comment by stefanie-hill** [comment]
I've got the exact same issues. Could someone please have a look?

**Comment by sel-aaraj** [comment]
I have the same issues. 
Someone has been able to explore more on this topic.

**Comment by machinedgod** [comment]
Just a note - I'm gonna try to fix this in my fork. If someone else has started, please let me know so I can jump to help instead.
